src/info/infra/repositories/DataverseInfoRepository.ts

```diff
@@ -0,0 +1,27 @@
+import { IDataverseInfoRepository } from '../../domain/repositories/IDataverseInfoRepository';
+import axios, { AxiosResponse } from 'axios';
+import { ReadError } from '../../../core/domain/repositories/ReadError';
+import { DataverseVersion } from '../../domain/models/DataverseVersion';
+
+export class DataverseInfoRepository implements IDataverseInfoRepository {
+  private apiUrl: string;
+
+  constructor(apiUrl: string) {
+    this.apiUrl = apiUrl;
+  }
+
+  public async getDataverseVersion(): Promise<DataverseVersion> {
+    const response = await axios.get(`${this.apiUrl}/info/version`).catch((error) => {
+      throw new ReadError(error.response.status + error.response.data ? ': ' + error.response.data.message : '');
```

Comment:
I think we'll need something like` await new Api.get(`${this.apiUrl}/info/version`) `because we don't want to be doing this check `error.response.status + error.response.data ? ': ' + error.response.data.message : '') `everytime we do an API call 

something like:


```
const response = await new Api.get(`${this.apiUrl}/info/version`).catch((error) => {
       throw new ReadError(error.message);
})

```

Reply:
Yes, something like a data source/API client object for the repository, encapsulating API requests and error handling. In any case, I already had these improvements in mind to avoid duplication and improve error handling (which is now very basic), as we add more API operations.